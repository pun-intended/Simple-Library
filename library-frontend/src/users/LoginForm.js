import {React, useContext, useState} from "react";
import {useNavigate} from "react-router-dom"
import {Form, FormGroup, Label, Input, Container, Button} from "reactstrap"
import AlertContext from "../AlertContext";

const LoginForm = ({login}) => {
    
    const INITIAL_STATE = {
        id: "",
        password: ""
    }
    const navigate = useNavigate()
    const {addAlert} = useContext(AlertContext)

    const[formData, setFormData] = useState(INITIAL_STATE)

    const handleSubmit = async (evt) => {
        evt.preventDefault()
        try{
            await login(formData)
            console.log(localStorage.token.length > 2)
            console.log(localStorage.token.length)
            console.log(localStorage.token)
            setFormData(INITIAL_STATE)
            if(localStorage.token.length > 2){
                addAlert("Welcome Back")
                navigate("/")
            }
        } catch(e){
            console.debug(e)
        }

    }

    const handleChange = evt => {
        const { name, value } = evt.target
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    }
    return(
    <Container className="LoginForm" >
        <Form onSubmit={handleSubmit} >
            <FormGroup > 
                <Label for="id">ID</ Label>
                <Input id="id"
                    name="id"
                    placeholder="ID"
                    type="text"
                    onChange={handleChange} 
                    value={formData.id}
                    />
                <Label for="password">Password</Label>
                <Input id="password"
                    name="password"
                    placeholder="Password"
                    type="password"
                    onChange={handleChange} 
                    value={formData.password} />
            </FormGroup>
            <Button>Login</Button>
        </Form>
    </Container>
    )
}

export default LoginForm;