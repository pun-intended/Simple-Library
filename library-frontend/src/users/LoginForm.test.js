import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from 'react-router-dom'

const login = jest.fn()
// const useNavigate = jest.fn()
// Smoke test
it("renders without crashing", () => {
    render(<MemoryRouter><LoginForm login={login} /></MemoryRouter>);
})

// Snapshot
it("matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><LoginForm login={login} /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// submit calls login method


