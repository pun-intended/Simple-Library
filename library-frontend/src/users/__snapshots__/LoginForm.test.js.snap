// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm container"
  >
    <form
      class=""
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="id"
        >
          ID
        </label>
        <input
          class="form-control"
          id="id"
          name="id"
          placegolder="ID"
          type="text"
          value=""
        />
        <label
          class="form-label"
          for="password"
        >
          Password
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          placegolder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        class="btn btn-secondary"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
